tools/api.py

```diff
@@ -1057,9 +1064,12 @@ def add_cpp_class(self) -> None:
     ${output_decls}
     if (${fname} == nullptr)
         throw std::runtime_error("${name} function is missing.");
+    std::array<char, 256> exception_msg;
     auto api_error_result = ${fname}(${args});
-    if (api_error_result != ${success})
-        throw std::runtime_error("Error in ${name}.");
+    if (api_error_result != ${success}) {
+        const std::string exception_str(exception_msg.data(), 256); 
```

Comment:
This will make a string that is always size 256, which may included extra uninitialized or null values. Have you looked at the output from this?

It would be nice to have a test case for this. I think the easiest way would be to have a hidden api mode to disable exception catching in the C api so we can catch the exception and check if the message matches:

```cpp
static thread_local bool disable_exception_catch = false;

// We dont put this file in migraphx.hpp since its only for testing
extern "C" void migraphx_test_private_disable_exception_catch(bool b)
{
    disable_exception_catch = false;
}

namespace migraphx {

template <class F>
migraphx_status try_(F f, bool output = true) // NOLINT
{
    if (disable_exception_catch)
    {
        f();
    }
    else
    {
        try
        {
            f();
        }
        catch(const migraphx::exception& ex)
        {
            if(output)
                std::cerr << "MIGraphX Error: " << ex.what() << std::endl;
            if(ex.error > 0)
                return migraphx_status(ex.error);
            else
                return migraphx_status_unknown_error;
        }
        catch(const std::exception& ex)
        {
            if(output)
                std::cerr << "MIGraphX Error: " << ex.what() << std::endl;
            return migraphx_status_unknown_error;
        }
        catch(...)
        {
            return migraphx_status_unknown_error;
        }
    }
    return migraphx_status_success;
}
...
} // namespace migraphx
```

So in the test you could add the operator with an incorrect shape and use `test::throws` to check for the exception and error message:

```cpp
TEST_CASE(run_sigmoid_custom_op_incorrect_shape)
{
  migraphx::program p;
  ...

  EXPECT(test::throws([&] {
           m.add_instruction(migraphx::operation("sigmoid_custom_op"), {x, alloc});
         }),
         "Expected error message");
}
```

Reply:
>. Have you looked at the output from this? This should just do const std::string exception_str(exception_msg.data()). 

I had looked at the output, It was working allright, but yes, i need to add null character. 

Added method to disable exception and added a test for exception message.